src/functional/program_processors/codegens/gtfn/gtfn_ir.py

```diff
@@ -179,7 +164,10 @@ class TagDefinition(Node):
 class FencilDefinition(Node, ValidatedSymbolTableTrait):
     id: SymbolName  # noqa: A003
     params: list[Sym]
-    function_definitions: list[Union[FunctionDefinition, ScanPassDefinition]]
+    function_definitions: list[
+        Union[FunctionDefinition, ScanPassDefinition, ImperativeFunctionDefinition]
```

Comment:
Wondering if we can find a way to separate the 2 fencil kinds, then we could also have one ir with reduce and one without. Maybe we can talk with @egparedes next week.

Reply:
@havogt @mroethlin, what was exactly the question here? I briefly looked at the code but I don't think I can follow the discussion with a bit more of context...